python/openassetio/hostAPI/Manager.py

```diff
@@ -1177,27 +1240,15 @@ def register(self, stringData, targetEntityRef, entitySpec, context, metadata=No
         ## @todo ... but conversely, setMetadata doesn't allow that data to be versioned
         ## This needs revisiting, as its not even really 'metadata' as we encourage
         ## hosts to treat it as first-class asset data.
-        entityRef = self.__impl.register(
-            stringData, targetEntityRef, entitySpec, context, self.__hostSession)
-        if metadata and entityRef:
-            self.__impl.setEntityMetadata(
-                entityRef, metadata, context, self.__hostSession, merge=True)
```

Comment:
[question] This is inconsistent with what `setEntityMetadata` expects and is thus broken. We expect a single `metadata` dict to apply to every entity. Here we're passing a list of metadata dicts, one for each entity. 

Which is correct? Or do we ignore this (and remove `setEntityMetadata` sooner rather than later)?

Reply:
Grr, I commented on the apparently "removed" lines (weren't removed, just slightly changed), but the implementation is almost exactly the same in the "added" lines. 

Interestingly, the old `registerMultiple` didn't have an option for `metadata`, so perhaps the simple solution to this is just to remove that parameter.

Reply:
Also see 14f675a where I added a missing test for this (hence discovering the problem) and had to comment it out with a `TODO`.

Reply:
> [question] This is inconsistent with what setEntityMetadata expects and is thus broken. We expect a single metadata dict to apply to every entity. Here we're passing a list of metadata dicts, one for each entity.

I would imagine, in the brave new world, then you would have a list of dicts. We can just fix this when we fix #4

Reply:
Fixed in f8e5a41 then added missing tests for the `IndexError` checks for `register` and `preflight` in 8553162 and e61e046, respectively.